.github/workflows/publish_charm.yaml

```diff
@@ -10,32 +10,101 @@ on:
+        id: get-branch-protection
+        env:
+          GITHUB_TOKEN: ${{ secrets.REPO_ACCESS_TOKEN }}
+        with:
+          route: GET /repos/{owner}/{repo}/branches/{branch}/protection/required_status_checks
+          repo: ${{ github.event.repository.name }}
+          owner: ${{ github.event.repository.owner.login }}
+          branch: main
+      - run: |
+          if [ ${{ fromJson(steps.get-branch-protection.outputs.data).strict }} != "true" ]; then
+            echo "::error::Strict checks are not enabled for this repository"
+            exit 1
+          fi
   get-runner-image:
     name: Get runner image
     uses: ./.github/workflows/get_runner_image.yaml
     with:
       working-directory: ${{ inputs.working-directory }}
+  release-charm-libs:
+    name: Release charm libs
+    needs: [ branch-up-to-date-check-enabled ]
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - if: inputs.working-directory != './'
+        name: Change directory
+        run: |
+          TEMP_DIR=$(mktemp -d)
+          cp -rp ${{inputs.working-directory}}/. $TEMP_DIR
+          rm -rf .* * || :
+          cp -rp $TEMP_DIR/. .
+          rm -rf $TEMP_DIR
+      - uses: canonical/charming-actions/release-libraries@2.3.0
+        name: Release libs
+        with:
+          credentials: ${{ secrets.CHARMHUB_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  get-commit-sha:
+    name: Get commit SHA
+    needs: [ branch-up-to-date-check-enabled ]
+    runs-on: ubuntu-22.04
+    outputs:
+      sha: ${{ env.LAST_COMMIT_SHA }}
+      wf-path: ${{ env.WF_PATH }}
+      wf-status: ${{ env.WF_STATUS }}
+    steps:
+      - name: Setup Environment (PR)  
+        if: ${{ github.event_name == 'pull_request' }}  
+        shell: bash  
+        run: |
+          echo "WF_STATUS=in_progress" >> ${GITHUB_ENV}
+          echo "WF_PATH=.github/workflows/workflow_test.yaml" >> ${GITHUB_ENV}
+          echo "LAST_COMMIT_SHA=${{ github.event.pull_request.head.sha }}" >> ${GITHUB_ENV}  
+      - name: Setup Environment (Push)  
+        if: ${{ github.event_name == 'push' }}  
+        shell: bash  
+        run: |
+          echo "WF_STATUS=success" >> ${GITHUB_ENV}
+          echo "WF_PATH=.github/workflows/integration_test.yaml" >> ${GITHUB_ENV}
+          echo "LAST_COMMIT_SHA=${GITHUB_SHA}" >> ${GITHUB_ENV}
+  get-images:
+    name: Get images
+    runs-on: ubuntu-22.04
+    outputs:
+      images: ${{ env.IMAGES }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get images
+        working-directory: ${{ inputs.working-directory }}
+        run: |
+          lines=$(find . -type f -name rockcraft.yaml | wc -l)
+          if [ $lines -ne 0 ]; then
+            IMAGES=$(find . -type f -name rockcraft.yaml | xargs -l yq '.name' | jq -Rsc '. / "\n" - [""]')
+          else
+            IMAGES=$(ls *.Dockerfile 2> /dev/null | sed s/\.Dockerfile// |  jq -Rsc '. / "\n" - [""]')
```

Comment:
This expects all Dockerfile to be in the working-directory, is this our "convention" ?